src/data.c

```diff
@@ -0,0 +1,68 @@
+#include <stdlib.h>
+#include <string.h>
+#include "data.h"
+
+/* Função que cria um novo elemento de dados data_t e que inicializa 
+ * os dados de acordo com os argumentos recebidos, sem necessidade de
+ * reservar memória para os dados.	
+ * Retorna a nova estrutura ou NULL em caso de erro.
+ */
+struct data_t *data_create(int size, void *data){
+    
+    if (size <= 0 || data == NULL)
+        return NULL;
+
+    struct data_t* d = malloc (sizeof(struct data_t));
+
+    d->data = data;
+    d->datasize = size;
+
+    return d;
+}
+
+/* Função que elimina um bloco de dados, apontado pelo parâmetro data,
+ * libertando toda a memória por ele ocupada.
+ * Retorna 0 (OK) ou -1 em caso de erro.
+ */
+int data_destroy(struct data_t *data){
+
+    if(data == NULL)
+        return -1;
+
+    free(data);    
+
+    return 0;
+}
+
+/* Função que duplica uma estrutura data_t, reservando a memória
+ * necessária para a nova estrutura.
+ * Retorna a nova estrutura ou NULL em caso de erro.
+ */
+struct data_t *data_dup(struct data_t *data){
+
+    if(data == NULL || data->data == NULL || data->datasize <= 0)
```

Comment:
Just as a comment, you can leave it as it is, but it's debatable whether data->data has to be different from NULL.

Reply:
Again, there's nothing to correct, just a comment